packages/liveblocks-client/src/client.ts

```diff
@@ -181,6 +181,10 @@ function getThrottleDelayFromOptions(options: ClientOptions): number {
 function prepareAuthentication(clientOptions: ClientOptions): Authentication {
   // TODO: throw descriptive errors for invalid options
   if (typeof clientOptions.publicApiKey === "string") {
+    if (clientOptions.publicApiKey.startsWith("sk_"))
```

Comment:
Wasn't sure wether to use `ifError` instead, let me know if it should be updated

Reply:
No, this is fine I think. `ifError` will only error in development mode, but if you use an `sk_` key in production in a public key position, that will never work anyway, so better to catch it in the client, always, even (especially?) in production.

Reply:
I'd probably catch both positive and negative cases here, aka structure it like:

```ts
const { publicApiKey, authEndpoint } = clientOptions;

if (publicApiKey !== undefined) {
  // A-ha! They're trying to use a public key! Let's check a few more things then…

  // First of all, ensure that they're not *also* setting the authEndpoint key!
  if (authEndpoint !== undefined) {
    // Throw friendly error explaining you cannot use both, and need to chose either of the two
  }

  if (typeof publicApiKey !== 'string') {
    // Throw friendly error that it should be a string, not whatever they used. Perhaps also hint that they should look for a key starting with `pk_` (and provide a link to the dashboard in there?)
  }

  if (startsWith('sk_')) {
    // A-ha! User provided the *secret* key! A common mistake! Give a friendly/helpful error message that they need to use the other one (and provide a link to the dashboard in there?)
  } else if (!publicApiKey.startsWith('pk_')) {
    // This is neither a public or a secret key, but another string. Tell them to look for their `pk_` key (and provide a link to the dashboard in there?)
  }

  // We made it! By now this is a (syntactically) valid PK, and we can continue
}
```

Reply:
Great comment, indeed, might as well make it bulletproof, all these use cases will end up happening at some point

Reply:
@nvie I've addressed your comments, let me know if this works 🙏